Fix the issue that invalid instance name (empty or null) can be created

zousheng opened the pull request:
Fix the issue https://github.com/mesos/mr-redis/issues/63

Commit:
Fix the issue that invalid instance name can be created when clicking the 'CREATE' buttong without focusing on any field

Commit:
Change sched/httplib/httplib.go to block invalid instance name null and spaces from front end and rest api request.

Commit:
Change redis_proxy.go to support loading redis instances from zk dynamically once redis is created or deleted, and generate local unique proxy listening port which stores in zk.

Commit:
Change CleanUpInterval and SyncZKIntervalSecs as 3 secs to keep the redis information consistent

Commit:
Fix issue that local_proxy on different machine might have different local port mapping to real server

Commit:
Change default log module to github.com/hhkbp2/go-logging for more specific logs, and remove clean_proxy_port func to avoid concurrent read and write map

dhilipkumars commented:
@zousheng Thanks for the PR? 

could you run 
`gofmt`
`golint`
and
`go vet`
in the code? we could get a cleaner new code. 

Also are you using redis-proxy in your project?

Commit:
Chagne broken pipe bug

Commit:
Change tcp proxy to fix connection reset error

Commit:
remove configs

Commit:
Use gofmt reformat the code

Commit:
remove panic and add zk path updats watch when redis created or removed

Commit:
Fix creating redis instance name empty issue.

Commit:
Fix master failover not working issue

Commit:
Add rand.Seed() to ensure the random port choosen for local proxy.

zousheng commented:
Sure,  I tried to use it in our production, and found few issues
1) I changed the proxy part to support proxy multiple redis servers, and store the mappings in the zk. In production the broken pipe issue occurred, and I added one io try catch for io.copy() method. and add the test-on-borrow on client, then it seems solved for now

2) In test environment, we tried to shutdown machine first to test the failover, it turns out master-slave failover not working, it's only working when I stop or kill the container, so this is very critical issue. Haven't checked the code yet. Would you please give some advice? Thanks
